### PR TITLE
Revert using SDL_Image for PNG screenshots

### LINKED ISSUE #752: PNG screenshots broken on OS X; 32 bit true colour on other platforms
@linguica reported that with the sdl2-branch, screenshots are coming out at 32bit (so >100KB each). I've confirmed this (and double checked that on OSX  at least they're correctly 8bit). I'm attempting a revert of the SDL2_Image commit to see if the older libpng code suffers this problem,but haven't coerced MSVC to link libpng12 properly yet.

### PR DESCRIPTION
**Pull request by jmtd**
This reverts commits 901ee2fe7588fd9dbe62f40b187bcaf364c0fce1 and
07afb7749ff807118431ea356ab56364ef8cfee8.

This change broke PNG screenshots on OS X and resulted in 32bit PNGs
on other platforms (large file sizes). The former might be a SDL_Image
bug, the latter definitely is.

Fixes #752

### CODE REVIEW

**Comment by fragglet** [comment]
Looks good. I'm not sure depending on SDL_image buys us anything anyway (since it's only PNGs we really care about).

**Comment by fabiangreffrath** [comment]
@jmtd Hi there, now that we have 2018 and Choco 3.0 and SDL2_image 2.0.2 have been released, could you please test if
- taking screenshots on OSX is still an issue
- screenshots are saved as paletted images
so that we can finally revert back to using SDL2_image?

**Comment by jmtd** [comment]
Hey folks: I can test on a Mac indeed, but I am shortly going to be returning my MBP (I've moved to a Linux/Thinkpad for work). But I haven't quite done it yet. So to be clear, I'm testing a rebase of those two patches on top of modern chocolate-doom v3? Is it still useful to test 32 bit powerpc?

**Comment by jmtd** [comment]
obvious tests first. current master PNG on Linux (Debian) works fine (8 bit 1600x1200, 66K or 37K 320x200 8 bit).

I've pushed a rebase of those two commits to https://github.com/jmtd/chocolate-doom/tree/fabian-sdl-image

**Comment by jmtd** [comment]
Ah SDL_Image output is not 8 bit , that's the reason (file says "8-bit/color RGBA")

Someone else pointed out that it looks like the SDL_Image is offset from the libpng version too: if I switch browser tabs it looks like it is, but I don't know if that's proof enough

https://jmtd.net/tmp/libpng_1600x1200.png
https://jmtd.net/tmp/sdl_image_1600x1200.png

**Comment by jmtd** [comment]
Sorry I'm being thick, need to confirm SDL image version too

**Comment by jmtd** [comment]
Should the version of SDL_Image in Debian sid work better? I see it carries your patch but I'm not seeing smaller screenshots after rebuilding against it

**Comment by jmtd** [comment]
OS X was fine (forgot about my wife's laptop, biult it on that). 

https://jmtd.net/tmp/choco_osx_1600x1200.png

https://jmtd.net/tmp/choco_osx_320x200.png

**Comment by fabiangreffrath** [comment]
Sorry, I find this a bit hard to follow.

Please make sure to check against **SDL2_Image 2.0.2** built **with libpng support**.

**Comment by fabiangreffrath** [comment]
I guess the images get larger for corrected aspect ratios, because the buffer that we scale our framebuffer to is a 32-bit RGBA buffer.

**Comment by jmtd** [comment]
OK tested properly on OS X. Sadly the outputs are still corrupted

https://jmtd.net/tmp/choco_osx_1600x1200.png
https://jmtd.net/tmp/choco_osx_320x200.png

**Comment by fabiangreffrath** [comment]
Ah, dammit, so there's still no way back... :(

**Comment by jmtd** [comment]
I guess the OS X issue needs fixing first. This was a separate laptop from the one I hit it in originally so that's useful but it would be nice for another OS X user to confirm it.
